src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,139 @@
+            }
+            String dateOn = input.substring(9).trim();
+            DateTimeFormatter format = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+            LocalDate date;
+            try {
+                date = LocalDate.parse(dateOn, format);
+            } catch (DateTimeParseException e) {
+                throw new DukeException("\tDate should be in format dd/mm/yyyy");
+            }
+
+            return new TasksOnCommand(date);
+        } else if (!input.isBlank()) {
+            Task newTask;
+            String desc;
+            DateTimeFormatter format = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+            LocalDateTime time;
+
+            if (input.startsWith("todo")) {
+                if (input.length() == 4 || input.substring(5).isBlank()) {
+                    throw new DukeException("\tThe description of a todo cannot be empty.");
+                }
+                desc = input.substring(5).trim();
+                newTask = new Todo(desc);
+            } else if (input.startsWith("deadline")) {
+                String[] components = input.split(" /by ");
+                if (components[0].length() == 8 || components[0].substring(9).isBlank()) {
+                    throw new DukeException("\tThe description of a deadline cannot be empty.");
+                }
+                if (components.length == 1 || components[1].isBlank()) {
+                    throw new DukeException("\tThe date of a deadline cannot be empty.");
+                }
+                desc = components[0].substring(9).trim();
+                try {
+                    time = LocalDateTime.parse(components[1].trim(), format);
+                } catch (DateTimeParseException e) {
+                    throw new DukeException("\tDate should be in format dd/mm/yyyy hh:mm");
+                }
+                newTask = new Deadline(desc, time);
+            } else if (input.startsWith("event")) {
+                String[] components = input.split(" /at ");
+                if (components[0].length() == 5 || components[0].substring(6).isBlank()) {
+                    throw new DukeException("\tThe description of an event cannot be empty.");
+                }
+                if (components.length == 1 || components[1].isBlank()) {
+                    throw new DukeException("\tThe date of an event cannot be empty.");
+                }
+                desc = components[0].substring(6).trim();
+                try {
+                    time = LocalDateTime.parse(components[1].trim(), format);
+                } catch (DateTimeParseException e) {
+                    throw new DukeException("\tDate should be in format dd/mm/yyyy hh:mm");
+                }
+                newTask = new Event(desc, time);
+            } else {
+                throw new DukeException("\tI don't know what that means :-(");
+            }
+
+            return new AddCommand(newTask);
+        }
+
+        return new DoNothingCommand();
+    }
```

Comment:
Perhaps consider splitting this method into several methods and using switch-case statements? This could reduce the length and nesting of the method.

Reply:
Thanks for the suggestion!

Reply:
Remember if you are changing to switch case. There is no indentation for the case word.
![Capture](https://user-images.githubusercontent.com/47635075/92298644-d0211180-ef7d-11ea-8ee1-a284be5ab660.PNG)